### PR TITLE
Fixes admin buildmode stripping people naked yet again, adds GetAllContentsIgnoring

### PR DESCRIPTION
**Pull request by silicons**
title. GetAllContentsIgnoring accepts an ignore typecache. Separate proc so the main one isn't bogged down because in most cases you won't need it.
If the turf empty proc is called without an ignore typecache the new proc automatically returns to using the normal proc so it isn't bogged down by always checking for an ignore typecache when it doesn't need to.

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -440,7 +440,7 @@
 /turf/proc/empty(turf_type=/turf/open/space, baseturf_type, list/ignore_typecache, forceop = FALSE)
 	// Remove all atoms except observers, landmarks, docking ports
 	var/static/list/ignored_atoms = typecacheof(list(/mob/dead, /obj/effect/landmark, /obj/docking_port, /atom/movable/lighting_object))
-	var/list/allowed_contents = typecache_filter_list_reverse(GetAllContents(), ignored_atoms | ignore_typecache)
+	var/list/allowed_contents = typecache_filter_list_reverse(GetAllContentsIgnoring(ignore_typecache = ignore_typecache), ignored_atoms)
```
Comment (optimumtact) [inline code/game/turfs/turf.dm]: have you tested this? because I can't recall if byond will work properly without specifying all named args

**Comment by kevinz000** [comment]
tested as of a commit before the current one

**Comment by AnturK** [comment]
#32029

**Comment by AnturK** [comment]
@kevinz000

**Comment by kevinz000** [comment]
On it

**Pull request merged**